website/static/js/components/publicNodes.js

```diff
@@ -0,0 +1,344 @@
+'use strict';
+
+var $ = require('jquery');  // jQuery
+var m = require('mithril'); // exposes mithril methods, useful for redraw etc.
+var oop = require('js/oop');
+var $osf = require('js/osfHelpers');
+var iconmap = require('js/iconmap');
+var lodashFind = require('lodash.find');
+var mHelpers = require('js/mithrilHelpers');
+var Raven = require('raven-js');
+
+var MAX_PAGES_ON_PAGINATOR = 7;
+var MAX_PAGES_ON_PAGINATOR_SIDE = 5;
+var PROJECTS_PAGE_SIZE = 5;
+
+var _buildUrl = function(page, user, nodeType) {
+    var query = {
+        'page[size]': PROJECTS_PAGE_SIZE,
+        'page': page || 1,
+        'embed': ['contributors'],
+        'filter[public]': 'true',
+        'version': '2.2'
+    };
+
+    if (nodeType === 'projects') {
+        query['filter[parent]'] = 'null';
+    }
+    else {
+        query['filter[parent][ne]'] = 'null';
+        query.embed.push('parent');
+    }
+
+    return $osf.apiV2Url('users/' + user +  '/nodes/', { query: query});
+};
+
+function _formatContributors(item) {
```

Comment:
If we're not going to DRY this up now, then could you please create a follow-up ticket to do so?